README.md

```diff
@@ -7,17 +7,147 @@
 
 ## Introduction
 
-Make sure students understand the ins and outs of `fetch()` -- e.g., the options object, headers, etc.
+In this lab we're going to use `fetch` to get data from Github, fork a
+repository, and post issues to our forked repository.
 
-Walk students through signing up for a GitHub auth token (click avatar in upper right, click "Settings", "click "Personal access tokens" on the side bar, generate a new token)
+Using `fetch` to get remote data after using `XMLHttpRequest` is a
+little like riding in a convertible with the top down after spending
+your whole life walking like a sucker.
 
-Have students practice creating issues on a repository (it could even be _this_ repository)
+![mean girls](http://i.giphy.com/4CP58gxwbBy2Q.gif)
 
-For the tests, redefine a global fetch as a spy and make sure that the student's functions are calling the right endpoint(s)
+Getting data from the Github API with `fetch` is super simple. If we're
+just trying to `GET` some JSON, we can do this:
+
+```js
+fetch('https://api.github.com/repos/jquery/jquery/commits').
+  then(resp => resp.json() ).
+  then(json => console.log(json))
+```
+
+Keeping in mind that we can use the `json` method of the `Body` mixin to
+render our response as JSON, and that each `then` passes its return
+value to the next `then` as an argument.
+
+### Authentication Token
+
+https://developer.github.com/v3/oauth/#3-use-the-access-token-to-access-the-api
+
+Github's API uses [OAuth2](https://developer.github.com/v3/oauth/) for
+authorization. Setting up the full OAuth2 authorization code grant workflow is beyond
+the scope of this lab, but it is described well in the Github
+[docs](https://developer.github.com/v3/oauth/), and I highly recommend
+you give it a read.
+
+Fortunately, Github also allows you to generate your own personal
+authorization token that we can use to give us authorized access to the
+API.
+
+If you already have a personal token that you've been using to make API
+requests, you can keep using that one. Otherwise, you'll need to
+generate a new one.
+
+To start, go to https://github.com/settings/tokens and click "Generate
+new token." Name it something like "Learn.co", and check `repo` scope. Once you generate
+the token, make sure to copy and paste it somewhere, because once you
+leave that page, you won't be able to see it again.
+
+Using the token to [access the API](https://developer.github.com/v3/oauth/#3-use-the-access-token-to-access-the-api) is a simple matter of creating an `Authorization` header with our request.
+
+We need to provide our authorization token in order to list our own
+repositories with this API, so let's add our `Authorization` header
+(don't forget to assign your token to `const token`).
```

Comment:
Would it be okay to use [`dotenv()`](https://www.npmjs.com/package/dotenv) for this?

We can set up the boilerplate for the students and just walk them through it quickly.